recipe/conda_build_config.yaml

```
@@ -2,3 +2,14 @@ rust_compiler_version:
     - 1.69
 maturin:
     - 1.1
+
+# The pinnings below allow a matrix build with two libprotobuf versions and
+# replace migrators for libprotobuf 1.57 / 1.58.
+libgrpc:
+    - 1.58
+    - 1.56
```

Also [`libgrpc` is pinned globally at `1.57`]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/7927b71839489c008649b56902f588cf7caf0c8e/recipe/conda_build_config.yaml#L470-L471 ). Do we want to include this as well?

```suggestion
    - 1.57
    - 1.56
```

---

The entries in this matrix are:
- Global pinning (1.57) plus migrations (which make it 1.58)
- libgrpc matching the older libprotobuf pinning that we need
  - libgrpc 1.56 was originally pinned with libprotobuf 4.23.3, from what I see here: https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/621d4a7b30d7814eaf75f556f37d609f8e6d33fc

---

You'll need to zip and match libprotobuf with libgrpc and also libabseil; they're not interrelated on source-level, but since we now need to migrate even for protobuf patch versions, we've coupled them 1:1 as follows:

| abseil | grpc | protobuf |
|-|-|-|
| 20230125 | 1.54 | 3.21 |
| 20230125 | 1.56 | 4.23.3 |
| 20230802 | 1.57 | 4.23.4 |
| 20230802 | 1.58 | 4.24.3 |

---

Though I guess it would be conceivable to rebuild grpc 1.56 & protobuf 4.23.3 with newer abseil.

---

More details [here](https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075)

---

Thanks Axel! 🙏

Currently we are hoping none of this is needed and that it was an oversight `libprotobuf` was in `host`. So are now trying to add it to `build` only

Though it is possible that won't work, in which case we will go back to adding in another set of versions. We had a working solution later in the commits based on a custom migrator, which was more ergonomic. This is probably what we would go back to if we need to add these in version pinnings in